rstudio/rstudio
# Guessing C++ style based on `LinkingTo`

### Intent

Follow up to #10340, second part of #10337

### Approach

Before checking user preferences, this function is used to have an educated guess as to what cpp style is used in the project/package: https://github.com/rstudio/rstudio/blob/b26934b1de260556296c7e82a4aa20bc3db1640c/src/cpp/session/modules/SessionCodeTools.R#L2591

### Automated Tests

> Indicate whether this change includes unit tests or integration tests, or link a work item or pull request to add those tests to another repo. If the change cannot or will not be covered by a test, indicate why.

### QA Notes

- open a package project
- add `LinkingTo: Rcpp` in the DESCRIPTION
- `File > New File > New C++ file` should create a c++ file with Rcpp style, i.e. 

```cpp
#include <Rcpp.h>
using namespace Rcpp;
...
```

- replace by `LinkingTo: cpp11` in the DESCRIPTION
- `File > New File > New C++ file` should create a c++ file with cpp11 style, i.e.

```cpp
#include <cpp11.hpp>
... 
```

- Remove the `LinkingTo` declaration
- `File > New File > New C++ file` should create a c++ file that obeys the setting from `Options > Package > C / C++ > Use template for new C++ file:`


### Checklist
